backend/server.go

```diff
@@ -0,0 +1,183 @@
+package backend
+
+import (
+	rice "github.com/GeertJohan/go.rice"
+	"github.com/gomodule/redigo/redis"
+	"html/template"
+	"log"
+	"net/http"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+)
+
+type Data struct {
+	People []Person
+	Awards []string
+}
+
+type Person struct {
+	Name string `redis:"name"`
+	Score float64 `redis:"score"`
+}
+
+type App struct {
+	pool *redis.Pool
+	config Config
+	tmpl 	template.Template
+}
+
+func renderTemplate(w http.ResponseWriter, people []Person, tmpl template.Template) {
+	err := tmpl.ExecuteTemplate(w, "index.html", Data{
+		People: people,
+		Awards: awards,
+	})
+	checkError(w, err)
+}
+
+func checkError(w http.ResponseWriter, err error) {
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func (app *App) homeHandler(w http.ResponseWriter, r *http.Request) {
+	people, err := app.loadPeople()
+	checkError(w, err)
+
+	renderTemplate(w, people, app.tmpl)
+
+	awards = make([]string ,0)
+}
+
+func (app *App) addHandler(w http.ResponseWriter, r *http.Request) {
+	conn := app.pool.Get()
+
+	err := r.ParseForm()
+	checkError(w, err)
+
+	for key, values := range r.Form {
+		name := strings.SplitN(key, "-", 2)[1]
+		value := values[0]
+
+		if value == "" {
+			continue
+		}
+
+		byPoints, err := strconv.ParseFloat(value, 64)
+		checkError(w, err)
+
+		// check awards
+		currScore, err := redis.Float64(conn.Do("HGET", name, "score"))
+		checkError(w, err)
+		if currScore + byPoints >= 100 {
+			awards = append(awards, name)
+			byPoints -= 100
```

Comment:
What if current score is 99 and bypoints is 100, would that mean the users score would be reset to 99?

Reply:
The intention was that if the point goes over 100, 100 points are deducted and the amount of score that's over 100 are kept as is. It's like the participant has used 100 points for the award but the 99 point they've earned is still kept. So " if current score is 99 and bypoints is 100, would that mean the users score would be reset to 99?" - yes and that was my intention. Now that I understand better how 🔆💯The System💯🔆 works I realized it should be reset to 0 no matter what